### DISCUSSION TITLE
All-in-one HTML file - Reporting options

### DISCUSSION THREAD

**Discussion by svishp** [Lumieducation/Lumi#1538] [General]
Hi,

I am experimenting with Lumi application to create H5P content in my desktop. Creation, editing, previewing is all working fine. I use Moodle and H5P. Recently I am also experimenting with saving H5P as All in one HtML5 file from Lumi application. I could create and send to learners. With reporting feature. My learners sent me back the report files. Which I used to see the visual report. This is all fine, but I have a doubt. If the H5P content type I sent is not having a retry option. If once learner opens and attempts the content type and downloads the report. If the learner feels he/she did not do well he/she can refresh the page and re attempt and re download the report and send that report to me right? So can we avoid this or can we record all the attempts by that user?

Please help me understand this area. Of course we can always use SCORM. For good reports. I am just hopping if this reporting works in HTML option. That would be easy for distribution of content among students of an institution who can not afford to have an LMS or so.
Thanks for the help in advance.

Sorry for my long post.

**Comment by sr258** [comment]
There's no way that we can reliably store the state of an H5P if it's not integrated into the LMS. The only way of avoiding cheating is if every attempt is logged on a server and if you can see the old records. As Lumi generated bundles that do everything on the client side, we cannot avoid potential cheating.

If you use Moodle already, you can also use the H5P plugin or the new support for H5P that's integrated in the core. Then you can view only the first attempts to see how students did on their first try.

**Reply by svishp** [reply]
Thank you for a quick reply. Yes I do use in Moodle. But was exploring this HTML export option. Thanks much for the reply.

**Reply by sr258** [reply]
If you can use the regular h5p plugins, the only situation in which it makes sense to use the HTML export is if you want to use content types which aren't installed by your admins or which aren't compatible, because the one created with Lumi is in a more recent content type version than the one installed in your Moodle.

**Reply by DegrangeM** [reply]
A possibility would be to generate a random id to the exported html file, and to then use localstorage to register the attempt and prevent further try. But this can be easily cheated with an other browser / icognito mode / etc.

**Reply by sr258** [reply]
True, but probably more of an edge use case. It might also be interesting to keep the user state in local storage to resume with the last entered answers. But h5p-nodejs-library doesn’t support this yet and the question is how many people actually host the generated html files on the web and how many people use them from the local disk. Local storage doesn‘t work when the browser displays html files from the local filesystem.

**Reply by JPSchellenberg** [reply]
I just tested it and it would work to store something in localStorage, but my fear is that this creates a false sense of security as it is really easily bypassed as you mentioned. We are currently also thinking about a pure player as desktop app. There you could implement such behaviour.

My approach to this problem was to not grade these assignments in any way but to use it for monitoring the learn-progress. My students know that it would not make sense to cheat on these assignments as I would not be able to help them properly.

**Comment by svishp** [comment]
How can we store the reports ? Once we pull the .lumi files into Analytics and view report
